README
This is an example plugin for theming in Deyan Shell (http://community.elgg.org/pg/plugins/project/752334/developer/vabel/deyan-shell).

Just add it to your mod folder and be sure that Deyan Shell is enabled before you activate this theme. For changing the theme go to personalize>theme in your elgg site.

ChangeLog
2011-07-15 	Angel Gabriel	<angel.wrt@gmail.com> 

	*manifest.xml:	Version upgraded to 1.0.1 & now requires that tropical
		theme need to be before Deyan Shell in plugin's list.

	*views/default/themes/tropical/css/modules.php: 
	*views/default/themes/tropical/css/components.php:
	*views/default/themes/tropical/css/forms.php:
	*views/default/themes/tropical/css/layout.php:
	*views/default/themes/tropical/css/navigation.php:	Included
		all the changes in elgg core release 1.8.0b2

	*views/default/themes/tropical/css/core.php:
	*views/default/themes/tropical/css/grid.php:
	*views/default/themes/tropical/css/helpers.php:
	*views/default/themes/tropical/css/misc.php:	Erased; as the new
		disposition in Deyan Shell


views/default/themes/tropical/constants.php
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme tropical
 * @constants
 * This is an alternative/example theme for deyan shell. This file contains the constants wich will be passed to css files
 * You can declare any constant you want, and you would use it in the css like $vars['theme'][$constant]
 * 
 **/  
$imagesurl = elgg_get_site_url() . 'mod/' . deyan_get_theme_plugin($theme) . "/views/default/themes/$theme/images/";
$vars = array(

	/*********************
	      MAIN
	*********************/

	'background' 	=> "url($imagesurl\wallpaper.jpg)",
	'shadow' 	=> '#020',
	'border' 	=> '#b1bc37',


	/*********************
	      WINDOWS
	*********************/
	'window_background' 	=> '#629337',
	'window_main' 		=> '#cee07d',
	'window_module' 	=> 'rgba(100, 100, 100, .2)',



	/*********************
	      PANEL
	*********************/
	'panel_background' 	=> '#629337',


	/*********************
	      MISC
	*********************/
	'aviable' 	=> '#aea',
	'unaviable' 	=> '#555',
	'hover' 	=> '#fff',

	);

views/default/themes/tropical/css/reset.php
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme tropical
 * @css reset
 * 
 **/ 
?>

/* ***************************************
	RESET CSS
*************************************** */
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, font, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td {
	margin: 0;
	padding: 0;
	border: 0;
	outline: 0;
	font-weight: inherit;
	font-style: inherit;
	font-size: 100%;
	font-family: inherit;
	vertical-align: baseline;
}
body {
	background: <?php echo $vars['theme']['background']; ?> fixed;
}
img {
	border-width:0;
	border-color:transparent;
}
:focus {
	outline: 0 none;
}
ol, ul {
	list-style: none;
}
em, i {
	font-style:italic;
}
ins {
	text-decoration:none;
}
del {
	text-decoration:line-through;
}
strong, b {
	font-weight:bold;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}
caption, th, td {
	text-align: left;
	font-weight: normal;
	vertical-align: top;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: "";
}
blockquote, q {
	quotes: "" "";
}
a {
	text-decoration: none;
}


views/default/themes/tropical/css/window.php
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme tropical
 * This file holds the styles for the windows
 * 
 **/ 
?> 
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme tropical
 * @css window
 *
 * This file holds the styles for the windows
 * 
 **/ 
?> 
/*******************************
*       TITLE BAR(header)      *
/******************************/
.deyan-titlebar-right{
      float: right;
      width: 250px;
      text-align: right;
}

.deyan-titlebar-right .deyan-icon-close{
      background: #e55 url(<?php echo $vars['theme']['url']; ?>images/base.png);
      width: 30px;
      height: 20px;      
      opacity: .7;
      border-bottom: 1px solid <?php echo $vars['theme']['border']; ?>;
      border-left: 1px solid <?php echo $vars['theme']['border']; ?>;
      border-radius: 0 0 0 5px;
      color: #fff;
	text-align: center;
	font-size: 20px;
	font-weight: bold;
}

.deyan-titlebar-right .deyan-icon-close:hover{
	box-shadow: -1px 1px 5px #e00;	
	opacity: 1;
}
.deyan-titlebar-right .deyan-icon-close:before {
	content: "x";
}

.deyan-titlebar-middle{
      float: right;
      width: 490px;
      text-align: center;
      text-shadow: 0px 0px 5px #333;
      color: #fff;
      font-weight: bold;
      padding: 3px 0 0 0; 
      cursor: pointer;
}

.deyan-titlebar-left{
      width: 250px;
      text-align: left;
}

/*******************************
*       STATUS BAR(footer)     *
/******************************/
.elgg-page-default .elgg-page-footer > .elgg-inner  .elgg-menu-hz{
      float: right;
}

/* header replacement for dashboard & profile */
.deyan-header-replacement{
      margin-top: 40px;
}


views/default/themes/tropical/css/typography.php
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme glass
 * @css typography
 * 
 **/ 
?>

/* ***************************************
	Typography
*************************************** */
body {
	font-size: 80%;
	line-height: 1.4em;
	font-family: "Lucida Grande", Arial, Tahoma, Verdana, sans-serif;
}

a {
	color: #4690D6;
}

a:hover,
a.selected { <?php //@todo remove .selected ?>
	color: #555555;
	text-decoration: underline;
}

p {
	margin-bottom: 15px;
}

p:last-child {
	margin-bottom: 0;
}

pre, code {
	font-family: Monaco, "Courier New", Courier, monospace;
	font-size: 12px;
	
	background:#EBF5FF;
	color:#000000;
	overflow:auto;

	overflow-x: auto; /* Use horizontal scroller if needed; for Firefox 2, not needed in Firefox 3 */

	white-space: pre-wrap;
	word-wrap: break-word; /* IE 5.5-7 */
	
}

pre {
	padding:3px 15px;
	margin:0px 0 15px 0;
	line-height:1.3em;
}

code {
	padding:2px 3px;
}

.elgg-monospace {
	font-family: Monaco, "Courier New", Courier, monospace;
}

blockquote {
	line-height: 1.3em;
	padding:3px 15px;
	margin:0px 0 15px 0;
	background:#EBF5FF;
	border:none;
	
	-webkit-border-radius: 4px;
	-moz-border-radius: 4px;
	border-radius: 4px;
}

h1, h2, h3, h4, h5, h6 {
	font-weight: bold;
	color: #0054A7;
}

h1 { font-size: 1.8em; }
h2 { font-size: 1.5em; line-height: 1.1em; padding-bottom:5px}
h3 { font-size: 1.2em; }
h4 { font-size: 1.0em; }
h5 { font-size: 0.9em; }
h6 { font-size: 0.8em; }

.elgg-heading-site, .elgg-heading-site:hover {
	font-size: 2em;
	line-height: 1.4em;
	color: white;
	font-style: italic;
	font-family: Georgia, times, serif;
	text-shadow: 1px 2px 4px #333333;
	text-decoration: none;
}

.elgg-heading-main {
	float: left;
	max-width: 530px;
	margin-right: 10px;
}
.elgg-heading-basic {
	color: #0054A7;
	font-size: 1.2em;
	font-weight: bold;
}

.elgg-subtext {
	color: #666666;
	font-size: 85%;
	line-height: 1.2em;
	font-style: italic;
}

.elgg-text-help {
	display: block;
	font-size: 85%;
	font-style: italic;
}

.elgg-quiet {
	color: #666;
}

.elgg-loud {
	color: #0054A7;
}

/* ***************************************
	USER INPUT DISPLAY RESET
*************************************** */
.elgg-output {
	margin-top: 10px;
}

.elgg-output dt { font-weight: bold }
.elgg-output dd { margin: 0 0 1em 1em }

.elgg-output ul, ol {
	margin: 0 1.5em 1.5em 0;
	padding-left: 1.5em;
}
.elgg-output ul {
	list-style-type: disc;
}
.elgg-output ol {
	list-style-type: decimal;
}
.elgg-output table {
	border: 1px solid #ccc;
}
.elgg-output table td {
	border: 1px solid #ccc;
	padding: 3px 5px;
}
.elgg-output img {
	max-width: 100%;
}

views/default/themes/tropical/css/panel.php
<?php
/**
 * @package Deyan
 * @author Angel Gabriel
 * @web http://angelgabriel.tk
 * @mail angel.wrt@gmail.com
 *
 * @theme tropical
 * @css panel
 *
 * This file holds the styles of the panel(topbar)
 * 
 **/ 
?> 

/* ***************************************
	PANEL LEFT
*****************************************/
.deyan-panel-left {
	float: left;
}
.deyan-panel-left .deyan-panel-item {
	float: left;
}


/* ***************************************
	PANEL RIGHT
*****************************************/
.deyan-panel-right {
	float: right;
}
.deyan-panel-right .deyan-panel-item {
	float: right;
}

/* ***************************************
	PANEL ITEMS
*****************************************/
/* Normal links */
.deyan-panel-item .elgg-button-normal {
	padding:9px 13px 0 13px;
	float: left;
	text-decoration:none;
	display:block;
	font-weight:bold;
	position:relative;
	margin-left:0;
	color: white;
	border: none;
	border-left: 2px solid transparent;
	border-radius: 0;
	box-shadow: 0 0 0;
	height: 26px;
}
.deyan-panel-item .elgg-button-normal:hover {
	background: none;
	text-decoration:none;
}

/* Dropdown links */
.deyan-panel-item .elgg-button-dropdown {
	padding:9px 6px 0 6px;
	float: left;
	text-decoration:none;
	display:block;
	font-weight:bold;
	position:relative;
	margin-left:0;
	color: white;
	border: none;
	border-top: 2px solid transparent;
	border-left: 2px solid transparent;
	border-radius: 0;
	box-shadow: 0 0 0;
	height: 24px;
}
.deyan-panel-item .elgg-button-dropdown:hover {
	background: none;
	text-decoration:none;
}

.deyan-panel-item .elgg-button-dropdown.elgg-state-active {
	background: rgb(35, 85, 35);
	outline: none;
	color: #ccc;
	border-top:2px solid rgb(29, 79, 29);
	border-left:2px solid rgb(29, 79, 29);
	border-radius: 5px;
}

/* full size modules*/
.deyan-panel-module-full {
	background: rgba(0, 0, 0, .9);
	position: fixed;
	display: none;
	border: none;
	width: 100%;
	height: 110%;
	margin-bottom: -20px;
	
}

.deyan-panel-module-full-inner {
	background: <?php echo $vars['theme']['panel_background']; ?>;
	border: 3px solid <?php echo $vars['theme']['border']; ?>;
	border-radius: 5px;
	width: 800px;
	margin: 130px auto 0 auto;
	padding: 5px;
	text-shadow: 0px 0px 5px #000;
}
.deyan-panel-module-full-inner label {
	color: #fff;
}

/* small size modules*/
.deyan-panel-module-small {
	background: none;
	position: fixed;
	display: none;
	border: none;
	width: 320px;
	z-index: 9999;
	
}

.deyan-panel-module-small-inner {
	position: absolute;
	background: <?php echo $vars['theme']['panel_background']; ?>;
	border: 3px solid <?php echo $vars['theme']['border']; ?>;
	border-top: none;
	border-radius: 0 0 5px 5px;
	margin: 0 0 0 0;
	padding: 5px;
	width: 300px;
	text-shadow: 0px 0px 5px #000;
	box-shadow: 0px 10px 5px <?php echo $vars['theme']['shadow']; ?>;

}
.deyan-panel-module-small-inner label {
	color: #fff;
}

/* ***************************************
	USER MENU
*****************************************/
.elgg-menu-user {
	padding: 10px;
}

.elgg-menu-user a {
	display: block;
	color: #fff;
	text-shadow: 0px 0px 5px #fff;
	border-radius: 5px;
	padding: 5px;
}
.elgg-menu-user a:hover {
	
	text-decoration: none;
	font-weight: bold;
	background: #666;
	color: #000;
}
.elgg-menu-user li.elgg-state-selected > a {
	background: rgb(35, 35, 35);
	outline: none;
	color: #fff;
	font-weight: bold;
	border: none;
	border-top:2px solid rgb(29, 29, 29);
	border-left:2px solid rgb(29, 29, 29);
	border-radius: 5px;
}
#deyan-panel-module-id-user-menu {
	width: 170px;
}
#deyan-panel-module-id-user-menu .deyan-panel-module-small-inner {
	width: 150px;
}

/* ***************************************
	START MENU
*****************************************/
.elgg-menu-start {
	float: left;
}

.elgg-menu-start > li {
	float: left;
}

.elgg-menu-start > li > a {
	padding: 15px;
	color: #fff;
	margin: 10px;
	width: 80px;
	height: 80px;
	text-align: center;
	border: 1px solid <?php echo $vars['theme']['border']; ?>;
	border-radius: 5px;
}

.elgg-menu-start > li > a:hover {
	color: #000;
	background: <?php echo $vars['theme']['hover']; ?>;
	font-weight: bold;
	text-decoration: none;
}





/**********************************

 PANEL MODULES

***********************************/

/**********************************
Search mod
***********************************/
.deyan-panel-right .elgg-search {
	float: right;
	margin: 5px 10px 0 10px;
}
.deyan-panel-right .elgg-search input[type=text] {
	width: 230px;
}
.deyan-panel-right .elgg-search input[type=submit] {
	display: none;
}
.elgg-search input[type=text] {
	border-radius: 5px;
	
	border: 1px solid #333;
	border-bottom: 2px groove #333;
	color: white;
	font-size: 12px;
	font-weight: bold;
	padding: 3px 4px 3px 26px;
	background: #777 url(<?php echo elgg_get_site_url(); ?>_graphics/elgg_sprites.png) no-repeat 2px -718px;
}
.elgg-search input[type=text]:focus, .elgg-search input[type=text]:active {
	background-color: #eef;
	background-position: 2px -700px;
	border: 1px solid white;
	color: #0054A7;
}

.search-list li {
	padding: 5px 0 0;
}
.search-heading-category {
	margin-top: 20px;
	color: #666666;
}

.search-highlight {
	background-color: #bbdaf7;
}
.search-highlight-color1 {
	background-color: #bbdaf7;
}
.search-highlight-color2 {
	background-color: #A0FFFF;
}
.search-highlight-color3 {
	background-color: #FDFFC3;
}
.search-highlight-color4 {
	background-color: #ccc;
}
.search-highlight-color5 {
	background-color: #4690d6;
}


/********************************
 Messages
********************************/
.message.unread a {
	font-weight: bold;
}